### PR TITLE
Add fastlane + github workflow for building example app

### PR DESCRIPTION
**Pull request by jnewc**
* Adds parameterised github workflow for building example apps
 * Bonus: fixes issue with selected card network type

### CODE REVIEW

**Comment by jnewc** [comment]
Functional, but won't pass until the `feature/co-badged-cards` is merged in primer-io/primer-sdk-ios

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        uses: maxim-lobanov/setup-xcode@v1
+        with:
+          xcode-version: '15.0'
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: "3.2"
+          bundler-cache: true
+      - name: Build with fastlane
+        run: |
+          bundle exec fastlane ios build_example_app "scheme:${{ matrix.project.scheme }}" "project_path:${{ matrix.project.path }}"
```
Comment (NQuinn27) [inline .github/workflows/build.yml:39]: `matrix.project` has no scheme, so this runs `bundle exec fastlane ios build_example_app "scheme:" "project_path:Co-Badged Cards/SwiftUI/Co-Badged Cards Example.xcodeproj"`

Fastlane seems to deal ok with it, but flagging

**Comment by NQuinn27** [comment]
LGTM just the one comment

**Review by NQuinn27 — APPROVED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (jnewc) [reply .github/workflows/build.yml:39]: Oh good catch! It must be defaulting to the first app scheme or something. Will fix 👍

**Pull request merged**